libsolidity/interface/CompilerStack.cpp

```diff
@@ -533,10 +540,15 @@ bool CompilerStack::compile(State _stopAfter)
 				{
 					try
 					{
-						if (m_generateEvmBytecode)
-							compileContract(*contract, otherCompilers);
-						if (m_generateIR || m_generateEwasm)
+						if (m_viaIR || m_generateIR || m_generateEwasm)
 							generateIR(*contract);
+						if (m_generateEvmBytecode)
+						{
+							if (m_viaIR)
+								generateEVMFromIR(*contract);
+							else
+								compileContract(*contract, otherCompilers);
+						}
```

Comment:
If we add this here we'll have two places where the decision not to generate EVM bytecode can happen. First `enableEvmBytecodeGeneration()` might get called to set `m_generateEvmBytecode` while compiler options are being processed (CLI does not do this but `StandardCompiler` and `viaYul` selector in tests do). Then this newly added code might still disable it if you use the new flag.

I think this should be happening in a single place. We could remove `enableEvmBytecodeGeneration()` and move the decision logic from `isEvmBytecodeRequested()` (defined in `StandardCompiler.cpp`) here.

Reply:
I'm not sure I understand?

`viaIR` only means to switch the entire pipeline (as discussed in the issue). I don't see how this influences whether EVM bytecode is generated or not.

Reply:
ok, disregard my comment. I got this wrong. These are two different decisions. Whether to generate the EVM bytecode at all and whether to generate it specifically using the old codegen.